tests/unit/linalg/test_hypre_vector.cpp

```diff
@@ -65,6 +78,85 @@ TEST_CASE("HypreParVector I/O", "[Parallel], [HypreParVector]")
    remove((prefix + "INFO." + suffix).c_str());
 }
 
+TEST_CASE("HypreParVector Move Constructor", "[Parallel], [HypreParVector]")
+{
+   int world_size, rank;
+   MPI_Comm_size(MPI_COMM_WORLD, &world_size);
+   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+   int size_per_rank = 2;
+
+   HYPRE_BigInt glob_size = world_size*size_per_rank;
+   std::vector<HYPRE_BigInt> col = internal::GenerateColumnPartitioning(world_size,
+                                                                        rank, size_per_rank);
+
+   // Initialize vector
+   HypreParVector v1(MPI_COMM_WORLD, glob_size, col.data());
+   for (int i=0; i<size_per_rank; ++i)
+   {
+      v1(i) = rank*size_per_rank + i;
+   }
+
+   // Make a copy so we can compare it later
+   HypreParVector v1_copy(v1);
+
+   // Also save off the data pointer
+   double* v1_data = v1.GetData();
+
+   // Now move from the original and compare
+   HypreParVector v1_move(std::move(v1));
+
+   REQUIRE(v1.GetOwnership() == 0);
+   REQUIRE(v1_move.GetOwnership() == 1);
+   REQUIRE(v1_move.Size() == v1_copy.Size());
+   REQUIRE(v1_move.GlobalSize() == v1_copy.GlobalSize());
+
+   REQUIRE(v1_move.GetData() == v1_data);
+
+   // The full HypreParVectors should also be the same
+   v1_copy -= v1_move;
+   REQUIRE(InnerProduct(v1_copy, v1_copy) == MFEM_Approx(0.0));
+}
+
+TEST_CASE("HypreParVector Move Assignment", "[Parallel], [HypreParVector]")
+{
+   int world_size, rank;
+   MPI_Comm_size(MPI_COMM_WORLD, &world_size);
+   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+   int size_per_rank = 2;
+
+   HYPRE_BigInt glob_size = world_size*size_per_rank;
+   std::vector<HYPRE_BigInt> col = internal::GenerateColumnPartitioning(world_size,
+                                                                        rank, size_per_rank);
+
+   // Initialize vector
+   HypreParVector v1(MPI_COMM_WORLD, glob_size, col.data());
+   for (int i=0; i<size_per_rank; ++i)
+   {
+      v1(i) = rank*size_per_rank + i;
+   }
+
+   // Make a copy so we can compare it later
+   HypreParVector v1_copy(v1);
+
+   // Also save off the data pointer
+   double* v1_data = v1.GetData();
+
+   // Now move from the original and compare
+   HypreParVector v1_move;
+   v1_move = std::move(v1);
+
+   REQUIRE(v1.GetOwnership() == 0);
```

Comment:
Could this also check the size and global size of v1? This would be similar to what happens in the `test_vector.cpp`. If this is possible the same checks should be added to the move constructor test above.

Reply:
Because the underlying vector is set to `nullptr` per https://github.com/mfem/mfem/pull/2639#discussion_r744037793 the call to `GlobalSize` will segfault, but I added checks for `size() == 0` and `GetData() == nullptr`.